tendermint/src/lite/types.rs

```diff
@@ -107,46 +85,156 @@ pub trait Commit {
 /// The default implementation returns true, iff at least a third of the trusted
 /// voting power signed (in other words at least one honest validator signed).
 /// Some clients might require more than +1/3 and can implement their own
-/// TrustLevel which can be passed into all relevant methods.
+/// TrustThreshold which can be passed into all relevant methods.
 pub trait TrustThreshold {
+    fn is_enough_power(&self, signed_voting_power: u64, total_voting_power: u64) -> bool;
+}
+
+pub struct TrustThresholdFraction {
+    numerator: u64,
+    denominator: u64,
+}
+
+impl TrustThresholdFraction {
+    pub fn new(numerator: u64, denominator: u64) -> Self {
+        // TODO: should we make sure we got a proper fraction here?
+        Self {
+            numerator,
+            denominator,
+        }
+    }
+}
+
+// TODO: should this go in the central place all impls live instead? (currently lite_impl)
```

Comment:
Do we want to keep `types.rs` purely abstract / generic only? If so this should go into a different file/module.